### ISSUE TITLE
Missing plugins?

### ISSUE THREAD

**Issue by rpavlik** [cnr-isti-vclab/meshlab#516]
There are some plugins that aren't in the qmake file despite being in the meshlabplugins directory.

A few build or build with minor updates:

- filter_aging: the .pro file says "no longer needed" but it does build and work just fine.
- filter_bnpts: Same as above, though it looks more specialized
- edit_hole: Needed minor tweaks to build, but seems to be OK: successfully closed a hole in the stanford bunny
- edit_texture: Needed minor tweaks to build, segfaults (in TextureEditor::SelectFromModel()'s call to ResetLayout) when you try to use it.
- sampleplugins/io_m: built fine
- sampleplugins/sample_xmlfilter: built fine
- sampleplugins/sampleio: built fine

Others are outdated enough that they probably should just be removed unless their functionality is sufficiently useful to merit the effort to update them.

- edit_pickpoints
- edit_slice

Others I don't know at all:

- sampleplugins/sample_pclfilter: Requires PCL, has an unimplemented filterArity method

**Comment by mcallieri** [comment]
Hi… first of all, thanks for your work!
I'll give a quick answer on some of the filter I know

- filter_aging. it should work, but has Always been more a toy than a useful filter. It modify the geoemtry to add craters, cracks and similar "damages", but only works properly on models that have been hand-modeled, and only with a specific ratio between the size of triangles and the global size of the object. So, basically, never. 

- filter_bnpts:  it might still work, but it was used to create the input of an external software that is now outdated and not supported anymore. We have integrated the Screened Poisson merging, that now integrates and improves the functionality of that external tool

- edit_hole:  the filter, in theory, is useful. however, we just gave up on it, as it was nevere stable enough to use it in most of the situations. 

- sampleplugins… they are meant to stay in the code to show a "skeleton" of implementation. It might be that some of them may also be useful in the distribution. I'll check 

- edit_pickpoints. This is hard. The filter is really old, has been impleemnted by an external laboratory, and was used to create pointsets for some other tool they had. As this was one of the only way to create list of hand-picked points and save it to a file for external use, we always kept in the distribution :). It is old, partly outdated, and works erratically. We might exclude from the distribution, as now the double-click gives you back the coordinates in the  

- edit_slice: if I remember correctly, it was a filter to create multiple cut-through sections. We already have the single-cut filter in the distribution, and I kinda rememebre this being really old and buggy

For the rest, I'll take a look (hopefully, soon)

**Comment by rpavlik** [comment]
cool, thanks! My test has been the stanford bunny ply, which filter_aging and edit_hole both worked just fine on, go figure.

**Comment by Quapcio** [comment]
Hi there
I'm new here but I'm in big need for edit_pickpoints plugin so I'm simultaneosly doing 2 things - searching for latest version with working plugin and trying to figure out what's insinde and how to improve it ;)

**Comment by rpavlik** [comment]
Presumably a very old version (like 1.3.x maybe?) Does double-clicking a point give you the data you need, as suggested above? Alternately, you could try updating it to newer API. I haven't tried that with any (despite my numerous patches, I've managed to avoid writing much in the way of actual code for the project :stuck_out_tongue: )  The build system should be there - you would just add `meshlabplugins/edit_pickpoints` to the variable in the CMake gui named something like EXTRA_PLUGINS.

**Comment by aslze** [comment]
Hi,

10 votes for **Pick Points** here! (See #570 )

Some days ago I opened Meshlab to sample some points on a mesh, as I had done many times. But, mmm. Where is the PP button, am I too sleepy to spot it? I searched toolbars and menus. And no Pick Points. Aaarg!

Yes, I used to use that often to get coordinates from specific points in meshes and use the exported XML in other programs. The double clic workaround is not a good solution as it moves the view. and does not keep the points as landmarks in the scene. Actually, as you say, the Pick Points tool was not that great, but it still has no real substitute.
